# Construction of pYPK0_RPS19a_ScRKI1_TPI1 (pMEC1133)

[pYPKa_Z_RPS19a](pYPKa_Z_RPS19a.ipynb)

[pYPKa_E_TPI1](pYPKa_E_TPI1.ipynb) 

In [1]:
import pydna
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568, p468, p467

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 = pydna.read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_RPS19a = pydna.read("pYPKa_Z_RPS19a.gb")

In [6]:
gene_clone = pYPKa_A_ScRKI1 = pydna.read("pYPKa_A_ScRKI1.gb")

In [7]:
terminator_clone = pYPKa_E_TPI1 = pydna.read("pYPKa_E_TPI1.gb")

In [8]:
p  = pydna.pcr( p167, p567, promoter_clone)
g  = pydna.pcr( p468, p467, gene_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [9]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [10]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(705), Amplicon(875), Amplicon(688))

In [11]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [12]:
asm

Assembly:
Sequences........................: [5681] [705] [875] [688]
Sequences with shared homologies.: [5681] [705] [688] [875]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [7703]
Linear products..................: [7801] [7764] [7753] [7740] [7146] [7113] [6915] [6308] [6288] [2181] [1530] [1526] [98] [61] [50] [37]

In [13]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|705bp_PCR_prod|50
|                          \/
|                          /\
|                          50|875bp_PCR_prod|37
|                                            \/
|                                            /\
|                                            37|688bp_PCR_prod|61
|                                                              \/
|                                                              /\
|                                                              61-
|                                                                 |
 -----------------------------------------------------------------

In [14]:
result = candidate.synced(pYPK0)

The new construct should have cseguid ``` 79JiqIENZc7fgGgYo9WS44MLsBA ``` and 7703 bp.

In [15]:
print(len(result))
result.cseguid()

7703


79JiqIENZc7fgGgYo9WS44MLsBA

In [16]:
result.name = "pMEC1132"
result.description = "pYPK0_RPS19a_ScRKI1_TPI1tp"
result.stamp()

cSEGUID_79JiqIENZc7fgGgYo9WS44MLsBA

In [17]:
result.write("pYPK0_RPS19a_ScRKI1_TPI1.gb")

pYPK0_RPS19a_ScRKI1_TPI1.gb

###[Download](pYP0 ... .gb) genbank file.

In [18]:
import pydna

In [19]:
reloaded = pydna.read("pYPK0_RPS19a_ScRKI1_TPI1.gb")

In [20]:
reloaded.cseguid()

79JiqIENZc7fgGgYo9WS44MLsBA

In [21]:
reloaded.description

'pYPK0_RPS19a_ScRKI1_TPI1tp cSEGUID_79JiqIENZc7fgGgYo9WS44MLsBA_2015-07-15T12:37:22.779394'